Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -0,0 +1,324 @@
+            if (!m_defaultFileName.empty())
+            {
+                defaultPath /= AZ::IO::Path(m_defaultFileName);
+            }
+
+            preselectedFilePath = defaultPath.c_str();
+        }
+        // Otherwise, we need to find the proper absolute path based on the relative path
+        // that is stored
+        else
+        {
+            // The GetFullSourcePathFromRelativeProductPath API only works on product paths,
+            // so we need to append the product extension to try and find it
+            AZStd::string relativePath = m_currentFilePath.String() + m_productExtension;
+
+            AZStd::string fullPath;
+            bool fullPathIsValid;
+            AssetSystemRequestBus::BroadcastResult(
+                fullPathIsValid, &AssetSystemRequestBus::Events::GetFullSourcePathFromRelativeProductPath, relativePath, fullPath);
+
+            // The full source path asset exists on disk, so use that as
+            // the pre-selected file when the user opens the file picker dialog
+            if (fullPathIsValid)
+            {
+                preselectedFilePath = fullPath.c_str();
+            }
+            // GetFullSourcePathFromRelativeProductPath failed so the file doesn't exist on disk yet
+            // So we need to find it by searching in the asset safe folders
+            else
+            {
+                bool assetSafeFoldersRetrieved = false;
+                AZStd::vector<AZStd::string> assetSafeFolders;
+                AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                    assetSafeFoldersRetrieved, &AzToolsFramework::AssetSystemRequestBus::Events::GetAssetSafeFolders, assetSafeFolders);
+
+                if (!assetSafeFoldersRetrieved)
+                {
+                    AZ_Error("PropertyFilePathCtrl", false, "Could not acquire a list of asset safe folders from the database.");
+                    return QString();
+                }
+
+                // Find an asset safe folder that already has existing sub-path that
+                // would satisfy the relative path for the file
+                for (const AZStd::string& assetSafeFolder : assetSafeFolders)
+                {
+                    AZ::IO::Path assetSafeFolderPath(assetSafeFolder);
+                    AZ::IO::Path relativeFolderPath = assetSafeFolderPath;
+
+                    if (m_currentFilePath.HasParentPath())
+                    {
+                        relativeFolderPath /= m_currentFilePath.ParentPath();
+                    }
+
+                    QDir currentFilePathDirectory(relativeFolderPath.c_str());
+                    if (currentFilePathDirectory.exists())
+                    {
+                        auto absoluteFilePath = (assetSafeFolderPath / m_currentFilePath);
+                        preselectedFilePath = absoluteFilePath.c_str();
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
This could be updated to bind the assetSafeFolder variable to a `const AZ::IO::Path&`

You could also use `AZ::IO::SystemFile::IsDirectory` to check if a directory exist.

```suggestion
                for (const AZ::IO::Path& assetSafeFolderPath : assetSafeFolders)
                {
                    AZ::IO::Path relativeFolderPath = assetSafeFolderPath;

                    if (m_currentFilePath.HasParentPath())
                    {
                        relativeFolderPath /= m_currentFilePath.ParentPath();
                    }

                    if (AZ::IO::SystemFile::IsDirectory(relativeFolderPath.c_str()))
                    {
                        auto absoluteFilePath = (assetSafeFolderPath / m_currentFilePath);
                        preselectedFilePath = QString::fromUtf8(absoluteFilePath.c_str(), static_cast<int>(absoluteFilePath.size()));
                        break;
                    }
                }
            }
        }
```

Also can you provide an explanation of what the intent of this logic is for.

It is quite confusing as to what type of path is being gatherred
The `assetSafeFolderPath`  has the `m_currentFilePath` parent directory appended to it, to check for file existence
But then the `m_currentFilePath` path is being appended to the `assetSafeFolderPath` and copied over to the result QString variable.

It seems simpler to just append m_currentFilePath to assetSafeFolderPath and then perform a check for existence of a parent directory at the end.

```suggestion
                for (AZ::IO::Path candidateFilePath : assetSafeFolders)
                {
                    candidateFilePath /= m_currentFilePath;

                    if (AZ::IO::FixedMaxPath parentCandidatePath = candidateFilePath.ParentPath();
                        AZ::IO::SystemFile::IsDirectory(parentCandidatePath.c_str()))
                    {
                        preselectedFilePath = QString::fromUtf8(candidateFilePath.c_str(), static_cast<int>(candidateFilePath.size()));
                        break;
                    }
                }
            }
        }
```

Reply:
Yes sorry your suggestion is more straightforward, I can adopt that and improve the comment as well to better explain what is going on, thanks!